compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -811,6 +808,7 @@ abstract class GenJSCode extends plugins.PluginComponent
       (for {
         f <- classSym.info.decls
         if !f.isMethod && f.isTerm && !f.isModule
+        if !f.hasAnnotation(JSOptionalAnnotation)
```

Comment:
Here they only happen in `$init$` methods of traits, and those methods are dropped altogether (as well as calls to them).